doc/contribute/devtheory.rst

```diff
@@ -0,0 +1,185 @@
+
+OBB Tree Construction
+---------------------
+
+First, the OBB tree for each surface is built. This is built top down where the
+top level is an OBB that contains all the surface’s facets. Then the set of facets
+is split roughly in half and two new OBBs form around each of the new sets.
+This continues until each of the OBBs the level contains a single facet. A 2D
+example is given below where a single surface has been faceted and the
+orresponding OBB tree is created.
+
+..  image:: red-tree.png
+    :height: 300
+    :width:  500
+    :alt:    Image of OBB tree structure for a surface
+
+After OBB trees have been created for each surface in a geometry, they are
+joined together to create a complete OBB tree for a volume. See the example
+below where the green, red, and blue surfaces make up a volume.
+
+..  image:: vol-obb-tree.png
+    :height: 500
+    :width:  675
+    :alt:    Image of OBB tree structure for a volume
+
+Implicit Complement
+~~~~~~~~~~~~~~~~~~~
+
+What is the Implicit Complement & How is it formed?
+---------------------------------------------------
+
+The implicit complement is the space not defined by the CAD model. It is formed
+by looking at the sense(s) of each surface individually. A surface between two
+volumes has both a forward and reverse sense defined with respect to the
+volumes. A surface that has only one sense defined, whether it is forward or
+reverse, means that only one side of that surface has a volume defined. The
+collection of all surfaces in the geometry that only have one sense defined are
+the surfaces that create the implicit complement by changing the sense already
+defined.
+
+OBB Tree Construction
+---------------------
+
+The OBB tree construction for the implicit complement is very similar to the
+construction of OBB trees for general volumes. However, since the surfaces that
+make the implicit complement are all part of other volumes, the OBB trees for
+the surfaces are already known, and therefore the only construction step
+necessary is joining the trees to create a complete OBB tree for the implicit
+complement volume.
+
+Ray History
+~~~~~~~~~~~
+
+A particle can be either streaming or reflecting. If streaming, the particle
+may only cross a particular surface once. To ensure this a list (or history)
+of the surfaces that particle has crossed is maintained. If a surface encountered
+is reflecting, the particle does not cross that surface and is therefore not
+added to the history. If a particle is or becomes a reflecting particle, the
+history of crossed surfaces is cleared.
+
+Point in Volume
+~~~~~~~~~~~~~~~
+
+Given a volume entity handle, position, and ray direction (optional), the
+point_in_volume function will test if the point is inside or outside the given
+volume. It is assumed that the test volume exists and is known. Passing a
+direction vector to this function adds robustness and ensures consistent results.
+Otherwise, a random direction is used.
+
+Ray Fire
+~~~~~~~~
```

Comment:
yes that is correct about only being able to call it one volume at a time and probably a good thing to clarify or note here